src/components/ResultViewer.js

```diff
@@ -23,7 +24,12 @@ export class ResultViewer extends React.Component {
   static propTypes = {
     value: PropTypes.string,
     editorTheme: PropTypes.string,
+    ResultsTooltip: PropTypes.any,
   };
+  constructor() {
+    super();
+    this._tooltip = document.createElement('div');
```

Comment:
`componentDidMount` is indeed  better place to create the tooltip element. The reason it accepts function returning component not function returning actual DOM element is because I think that it looks cleaner from API consumer perspective - graphiql currently shield user from working with dom it does under the hood. With this API graphiql could potentially even change to something other that codemirror without breaking its API. But in the end I am indifferent to what the final solution is.